avro/src/test/java/com/fasterxml/jackson/dataformat/avro/jsr310/AvroJavaTimeModule_schemaCreationTest.java

```diff
@@ -0,0 +1,75 @@
+package com.fasterxml.jackson.dataformat.avro.jsr310;
+
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.dataformat.avro.AvroMapper;
+import com.fasterxml.jackson.dataformat.avro.schema.AvroSchemaGenerator;
+import org.apache.avro.LogicalType;
+import org.apache.avro.Schema;
+import org.apache.avro.specific.SpecificData;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(Parameterized.class)
+public class AvroJavaTimeModule_schemaCreationTest {
+
+    @Parameter(0)
+    public Class testClass;
+
+    @Parameter(1)
+    public Schema.Type expectedType;
+
+    @Parameter(2)
+    public String expectedLogicalType;
+
+    @Parameters(name = "With {0}")
+    public static Collection testData() {
+        return Arrays.asList(new Object[][]{
+                // Java type    | expected Avro type    | expected logicalType
+                {Instant.class, Schema.Type.LONG, "timestamp-millis"},
+                {OffsetDateTime.class, Schema.Type.LONG, "timestamp-millis"},
+                {ZonedDateTime.class, Schema.Type.LONG, "timestamp-millis"},
+                {LocalDateTime.class, Schema.Type.LONG, "local-timestamp-millis"},
+                {LocalDate.class, Schema.Type.INT, "date"},
+                {LocalTime.class, Schema.Type.INT, "time-millis"},
+        });
+    }
+
+    @Test
+    public void testSchemaCreation() throws JsonMappingException {
```

Comment:
One quick question here: since changes to schema visitor are applied regardless of module (I think?), would there be problems/changes to logic for existing users? That is, is there any possible backwards-compatibility concerns wrt creating more granular Avro schema types?
If there are, it might make sense to add configurability to allow disabling use of date/time types.

Reply:
Hmmm,  is there a way how to control such switch from a module ? 
I guess not.